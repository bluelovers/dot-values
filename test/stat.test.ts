/**
 * Created by User on 2020/1/12.
 */

import { get } from '..';

let testData = JSON.parse(JSON.stringify(require('./deep.json'))) as typeof import('./deep.json');

describe(`test * in object`, () =>
{
	it(`get a.*.c`, function ()
	{
		let actual = get(testData, 'a.*.c');
		let expected = [
			undefined,
			undefined,
			undefined,
			{
				"arr": [
					{
						"d": {
							"e": {
								"arr": [
									{
										"f": true,
									},
									{
										"f": true,
									},
								],
							},
						},
						"x": true,
					},
					{
						"d": {
							"e": {
								"arr": [
									{
										"f": false,
									},
									{
										"f": false,
									},
								],
							},
						},
					},
					{
						"d": {
							"e": {
								"arr": [
									{
										"f": false,
									},
									{
										"f": false,
									},
								],
							},
						},
					},
				],
			},
		];

		expect(actual).toStrictEqual(expected);
		expect(actual).toMatchSnapshot();
	});

	it(`get a.*`, function ()
	{
		//console.log('it:inner', currentTest.title);
		//console.log('it:inner', currentTest.fullTitle());

		let actual = get(testData, 'a.*');
		let expected = [
			{
				arr: [
					{
						d: {
							e: { arr: [{ f: true }, { f: true }] },
						},
						y: true,
					},
					{
						d: {
							e: { arr: [{ f: false }, { f: false }] },
						},
						y: true,
					},
					{
						d: {
							e: { arr: [{ f: false }, { f: false }] },
						},
						y: true,
					},
				],
			},
			{
				arr: [
					{
						d: {
							e: { arr: [{ f: true }, { f: true }] },
						},
						z: true,
					},
					{
						d: {
							e: { arr: [{ f: false }, { f: false }] },
						},
						z: true,
					},
					{
						d: {
							e: { arr: [{ f: false }, { f: false }] },
						},
						z: true,
					},
				],
			},
			{
				arr: [
					{
						d: {
							e: { arr: [{ f: true }, { f: true }] },
						},
						w: true,
					},
					{
						d: {
							e: { arr: [{ f: false }, { f: false }] },
						},
						w: true,
					},
					{
						d: {
							e: { arr: [{ f: false }, { f: false }] },
						},
						w: true,
					},
				],
			},
			{
				c: {
					arr: [
						{
							d: {
								e: { arr: [{ f: true }, { f: true }] },
							},
							x: true,
						},
						{
							d: {
								e: { arr: [{ f: false }, { f: false }] },
							},
						},
						{
							d: {
								e: { arr: [{ f: false }, { f: false }] },
							},
						},
					],
				},
			},
		];

		expect(actual).toStrictEqual(expected);
		expect(actual).toMatchSnapshot();
	});
});
