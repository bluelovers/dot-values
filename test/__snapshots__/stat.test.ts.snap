// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test * in object get a.* 1`] = `
[
  {
    "arr": [
      {
        "d": {
          "e": {
            "arr": [
              {
                "f": true,
              },
              {
                "f": true,
              },
            ],
          },
        },
        "y": true,
      },
      {
        "d": {
          "e": {
            "arr": [
              {
                "f": false,
              },
              {
                "f": false,
              },
            ],
          },
        },
        "y": true,
      },
      {
        "d": {
          "e": {
            "arr": [
              {
                "f": false,
              },
              {
                "f": false,
              },
            ],
          },
        },
        "y": true,
      },
    ],
  },
  {
    "arr": [
      {
        "d": {
          "e": {
            "arr": [
              {
                "f": true,
              },
              {
                "f": true,
              },
            ],
          },
        },
        "z": true,
      },
      {
        "d": {
          "e": {
            "arr": [
              {
                "f": false,
              },
              {
                "f": false,
              },
            ],
          },
        },
        "z": true,
      },
      {
        "d": {
          "e": {
            "arr": [
              {
                "f": false,
              },
              {
                "f": false,
              },
            ],
          },
        },
        "z": true,
      },
    ],
  },
  {
    "arr": [
      {
        "d": {
          "e": {
            "arr": [
              {
                "f": true,
              },
              {
                "f": true,
              },
            ],
          },
        },
        "w": true,
      },
      {
        "d": {
          "e": {
            "arr": [
              {
                "f": false,
              },
              {
                "f": false,
              },
            ],
          },
        },
        "w": true,
      },
      {
        "d": {
          "e": {
            "arr": [
              {
                "f": false,
              },
              {
                "f": false,
              },
            ],
          },
        },
        "w": true,
      },
    ],
  },
  {
    "c": {
      "arr": [
        {
          "d": {
            "e": {
              "arr": [
                {
                  "f": true,
                },
                {
                  "f": true,
                },
              ],
            },
          },
          "x": true,
        },
        {
          "d": {
            "e": {
              "arr": [
                {
                  "f": false,
                },
                {
                  "f": false,
                },
              ],
            },
          },
        },
        {
          "d": {
            "e": {
              "arr": [
                {
                  "f": false,
                },
                {
                  "f": false,
                },
              ],
            },
          },
        },
      ],
    },
  },
]
`;

exports[`test * in object get a.*.c 1`] = `
[
  undefined,
  undefined,
  undefined,
  {
    "arr": [
      {
        "d": {
          "e": {
            "arr": [
              {
                "f": true,
              },
              {
                "f": true,
              },
            ],
          },
        },
        "x": true,
      },
      {
        "d": {
          "e": {
            "arr": [
              {
                "f": false,
              },
              {
                "f": false,
              },
            ],
          },
        },
      },
      {
        "d": {
          "e": {
            "arr": [
              {
                "f": false,
              },
              {
                "f": false,
              },
            ],
          },
        },
      },
    ],
  },
]
`;
